crates/cxx-qt-gen/src/syntax/types.rs

```diff
@@ -119,6 +119,85 @@ pub fn extract_qobject_ident(ty: &Type) -> Result<(Ident, Option<Mut>)> {
     }
 }
 
+fn err_unsupported_type<T: quote::ToTokens>(ty: &T) -> Error {
+    Error::new_spanned(ty, "Type not supported by CXX-Qt!")
+}
+
+fn extract_generic_argument_lifetimes(argument: &GenericArgument) -> Result<Vec<Lifetime>> {
+    match argument {
+        GenericArgument::Lifetime(lifetime) => Ok(vec![lifetime.clone()]),
+        GenericArgument::Type(ty) => extract_lifetimes(ty),
+        _ => Err(err_unsupported_type(argument)),
+    }
+}
+
+fn extract_pathsegment_lifetimes(segment: &PathSegment) -> Result<Vec<Lifetime>> {
```

Comment:
wonder if this stuff could be in it's own lifetimes.rs under syntax as they are sortof types helpers but very specifically lifetime helpers ?

Reply:
Hm, good point, especially for those helpers :thinking:

Reply:
Done